modules/gdscript/editor/gdscript_translation_parser_plugin.cpp

```diff
@@ -383,10 +383,17 @@ void GDScriptEditorTranslationParserPlugin::_assess_call(const GDScriptParser::C
 			_add_id(p_call->arguments[1]->reduced_value, p_call->arguments[1]->start_line);
 		}
 	} else if (function_name == fd_add_filter) {
-		// Extract the 'JPE Images' in this example - get_node("FileDialog").add_filter("*.jpg; JPE Images").
-		if (!p_call->arguments.is_empty()) {
-			_extract_fd_filter_string(p_call->arguments[0], p_call->arguments[0]->start_line);
-		}
+    	// Extract filter name from FileDialog.add_filter()
+    	if (!p_call->arguments.is_empty()) {
+        	// Check if using the two-parameter format first
+        	if (p_call->arguments.size() >= 2 && _is_constant_string(p_call->arguments[1])) {
+            	// Second parameter is the filter name, extract it directly
+            	_add_id(p_call->arguments[1]->reduced_value, p_call->arguments[1]->start_line);
+        	} else {
+            	// Extract filter from single-parameter format
+            	_extract_fd_filter_string(p_call->arguments[0], p_call->arguments[0]->start_line);
+        	}
+    	}
```

Comment:
```suggestion
		if (p_call->arguments.size() == 1) {
			// The first parameter may contain a description, like `"*.jpg; JPEG Images"`.
			_extract_fd_filter_string(p_call->arguments[0], p_call->arguments[0]->start_line);
		} else if (p_call->arguments.size() >= 2) {
			// The second optional parameter can be a description.
			if (_is_constant_string(p_call->arguments[1])) {
				_add_id(p_call->arguments[1]->reduced_value, p_call->arguments[1]->start_line);
			}
		}
```

Also, we should probably fix `_extract_fd_filter_string()` since the description is optional, to avoid printing unnecessary errors.

https://github.com/godotengine/godot/blob/5a2ec25044f6ce31bb7aaf11229c6c02ec609670/modules/gdscript/editor/gdscript_translation_parser_plugin.cpp#L405-L411